### PR TITLE
Relocate sdc variant selection

### PR DESCRIPTION
**Pull request by ibaz-nordic**
This is the other part of https://github.com/nrfconnect/sdk-nrfxlib/pull/572

### CODE REVIEW

File: subsys/bluetooth/Kconfig [hunk 1]
```diff
@@ -17,9 +17,9 @@ config BT_MAX_CONN
 	range 1 20 if BT_LL_SOFTDEVICE
 	default 2 if BT_LL_SOFTDEVICE && BT_CENTRAL && BT_PERIPHERAL
 
-if BT_LL_SOFTDEVICE
+if !SOC_NRF5340_CPUAPP
```
Comment (rugeGerritsen) [inline subsys/bluetooth/Kconfig:20]: Should this be: `if BT_CTLR`?

File: subsys/bluetooth/controller/Kconfig [hunk 2]
```diff
@@ -4,6 +4,36 @@
...
+	# The SoftDevice Controller only supports nRF52 and nRF53, it does not support nRF51
+	# or nRF91 (nRF91 doesn't even have the physical HW needed for BLE).
+	depends on (SOC_SERIES_NRF52X || SOC_NRF5340_CPUNET)
+	help
+	  Use SoftDevice Link Layer implementation.
+
+endchoice
+
+if BT_LL_SOFTDEVICE
 menu "Nordic BLE controller"
```
Comment (rugeGerritsen) [inline subsys/bluetooth/controller/Kconfig:37]: When you are touching this file, you could also change this menu to "SoftDevice Controller"

File: subsys/bluetooth/Kconfig (hunk 1 above)
Comment (jori-nordic) [reply subsys/bluetooth/Kconfig:20]: yes, I think so

**Review by rugeGerritsen — APPROVED** [review]

File: doc/nrf/ug_multiprotocol_support.rst [hunk 3]
```diff
@@ -17,17 +17,7 @@ The multiprotocol solution available for Thread and Zigbee is dynamic.
...
-
-To enable the multiprotocol support for either Thread or Zigbee, set the following Kconfig option:
-
-* :kconfig:`CONFIG_BT_LL_SOFTDEVICE_DEFAULT`
-
-This option activates the :ref:`nrfxlib:softdevice_controller`, which by default supports the multiprotocol features.
-Disabling the SoftDevice Controller also disables the multiprotocol support.
-
-After enabling this option, the application can start configuring Bluetooth® LE's API options alongside Thread's or Zigbee's in the respective :file:`.conf` file.
+Multiprotocol operation is supported if using SoftDevice Controller (:kconfig:`CONFIG_BT_LL_SOFTDEVICE`), which is selected by default.
```
Comment (wiba-nordic) [inline doc/nrf/ug_multiprotocol_support.rst:20]: I think the information in this sentence should be incorporated to the sentence on line 13, or between lines 13 and 14, now that this isn't an entire section.

**Review by wiba-nordic — APPROVED** [review]
I think that's good.

File: subsys/bluetooth/controller/Kconfig [hunk 4]
```diff
@@ -86,13 +123,41 @@ config SDC_RX_STACK_SIZE
...
+						     BT_CTLR_PHY_CODED || \
+						     BT_CTLR_ADV_PERIODIC || \
+						     SOC_NRF5340_CPUNET)
+	default SOFTDEVICE_CONTROLLER_CENTRAL if BT_OBSERVER
+	default SOFTDEVICE_CONTROLLER_PERIPHERAL if BT_BROADCASTER
+	help
+	  Select a SoftDevice Controller variant.
 
-endchoice
+config SOFTDEVICE_CONTROLLER_PERIPHERAL
```
Comment (carlescufi) [inline subsys/bluetooth/controller/Kconfig:139]: Why? We already have `BT_PERIPHERAL` and `BT_CENTRAL`?

Comment (carlescufi) [reply subsys/bluetooth/controller/Kconfig:139]: @cvinayak @alwa-nordic @joerchan any thoughts here?

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/Kconfig (hunk 4 above)
Comment (rugeGerritsen) [reply subsys/bluetooth/controller/Kconfig:139]: @carlescufi . These are the names of the libraries. See the help text

Comment (jori-nordic) [reply subsys/bluetooth/controller/Kconfig:139]: I guess they're badly named, but they are three distinct libraries, with the central/peripheral being stripped-out size-optimized versions and the multirole being a full-featured version

Comment (carlescufi) [reply subsys/bluetooth/controller/Kconfig:139]: I see, fair enough.

Can we please maintain some consistency here? We chose `CTLR` a long time ago for the upstream one, can we try to do the same here? Maybe:

`BT_CTLR_SD_<>`

I mean you are not even prefixing with `BT_` here

Comment (jori-nordic) [reply subsys/bluetooth/controller/Kconfig:139]: I mean we could either rename or move the logic to cmake, but then we lose the ability to force one or the other. Maybe having a check on BT_CENTRAL and BT_PERIPHERAL, and exposing two choices: SIZE_OPTIMIZED and FULL_FEATURED would be better ?

**Review by carlescufi — APPROVED** [review]
Will be continued in a follow-up PR

File: subsys/bluetooth/controller/Kconfig (hunk 4 above)
Comment (carlescufi) [reply subsys/bluetooth/controller/Kconfig:139]: I don't mind either way, as long as we maintain consistency. Please add @tejlmand as a reviewer in the follow-up PR.

Comment (rugeGerritsen) [reply subsys/bluetooth/controller/Kconfig:139]: Created https://github.com/nrfconnect/sdk-nrf/pull/6107

**Pull request merged**